src/io-methods/s3.js

```diff
@@ -37,21 +38,37 @@ const s3 = new aws.S3({
 });
 
 exports.readManifest = function (filePath) {
-  return new Promise(function (resolve, reject) {
+  return new Promise(async function (resolve, reject) {
```

Comment:
Putting an async function as a promise constructor is unusual and should be avoided. Instead of returning a new promise, let's just change the readManifest function to be an async function. Doing so will simplify the code in a few ways, including not having to catch errors just so we can reject them.

Reply:
Made the changes.